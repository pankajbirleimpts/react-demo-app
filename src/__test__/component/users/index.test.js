import React from "react";
import { shallow } from "enzyme";
import UserList from "../../../components/users/index";
import { storeFactory } from "../../test/TestUtils";

const setup = props => {
  const store = storeFactory(props);
  const wrapper = shallow(<UserList store={store} />)
    .dive()
    .dive();
  return wrapper;
};

describe("UserList component", () => {
  const props = {
    user: {
      isLoading: false,
      allUsers: [
        {
          id: 4,
          firstName: "Pankaj",
          lastName: "Birle"
        },
        {
          id: 5,
          firstName: "Rohan",
          lastName: "Bajpai"
        }
      ]
    }
  };
  test("Snapshot of UserList component", () => {
    const wrapper = setup(props);
    expect(wrapper).toMatchSnapshot();
  });

  test("Check the isLoading in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.user.isLoading).toBe(props.user.isLoading);
  });

  test("Check the `getAllUsers` action in props", () => {
    const wrapper = setup(props);
    const componentProps = wrapper.instance().props;
    expect(componentProps.getAllUsers).toBeInstanceOf(Function);
  });
});
