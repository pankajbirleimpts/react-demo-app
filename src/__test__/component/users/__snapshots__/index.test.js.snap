// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList component Snapshot of UserList component 1`] = `
<Grid>
  <Loading
    isLoading={false}
  />
  <GridRow>
    <GridColumn
      textAlign="left"
      width={8}
    >
      <h3>
        Manage Users
      </h3>
    </GridColumn>
    <GridColumn
      textAlign="right"
      width={8}
    >
      <Link
        className="ui button primary"
        to="/add-user"
      >
        + Add User
      </Link>
    </GridColumn>
  </GridRow>
  <GridRow
    centered={true}
  >
    <Connect(UnconnectedCustomTable)
      columns={
        Array [
          Object {
            "keyName": "firstName",
            "label": "First Name",
            "searchable": true,
            "sort": true,
            "sortable": true,
          },
          Object {
            "keyName": "lastName",
            "label": "Last Name",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "email",
            "label": "Email",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "employeeId",
            "label": "Employee Id",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
          Object {
            "keyName": "balance",
            "label": "Balance",
            "searchable": true,
            "sort": null,
            "sortable": true,
          },
        ]
      }
      module="USERS"
      tableData={
        Array [
          Object {
            "firstName": "Pankaj",
            "id": 4,
            "lastName": "Birle",
          },
          Object {
            "firstName": "Rohan",
            "id": 5,
            "lastName": "Bajpai",
          },
        ]
      }
    />
  </GridRow>
</Grid>
`;
